lib/pymedphys/_experimental/streamlit/apps/sum_doses.py

```diff
@@ -28,56 +28,121 @@
+            * Patient Name: `{get_pretty_patient_name_from_dicom_dataset(datasets[0])}`
+            """
+        )
+
+    if len(datasets) < 2:
+        st.stop()
+
+    st.write("---")
+    st.write("Summing doses...")
+
+    ds_summed = sum_doses_in_datasets(datasets)
+    _save_dataset_to_downloads_dir(ds_summed)
+
+    st.write("Done!")
+    st.markdown(
+        "*Download the summed DICOM dose file from "
+        "[downloads/RD.summed.dcm](downloads/RD.summed.dcm)*"
+    )
+
+
+def _load_and_check_files_valid(
+    files: Sequence[BinaryIO],
+) -> List["pydicom.dataset.Dataset"]:
 
-    if not len(files) >= 2:
-        raise ValueError("`files` must contain at least 2 elements")
+    ds0 = _load_dicom_file(files[0])
+    datasets = [ds0]
 
-    datasets = []
-    ds0 = None
+    for fh in files[1:]:
+        ds = _load_dicom_file(fh)
 
-    for i, fh in enumerate(files):
         try:
-            ds = pydicom.dcmread(fh)
-        except pydicom.errors.InvalidDicomError:
-            st.error(f"'{fh.name}' is not a valid DICOM file")
-            return None
-
-        if not ds.Modality == "RTDOSE":
-            st.error(f"File '{fh.name}' is not a valid DICOM RT Dose file")
-            return None
-
-        if i == 0:
-            ds0 = ds
-        else:
-            if not ds.PatientID == ds0.PatientID:
-                st.error(
-                    f"File '{fh.name}' has a different DICOM Patient "
-                    f"ID from '{files[0].name}'"
-                )
-                return None
-            if not ds.DoseUnits == ds0.DoseUnits:
-                st.error(
-                    f"File '{fh.name}' has a different value for "
-                    f"DoseUnits ({ds.DoseUnits}) from "
-                    f"'{files[0].name}' ({ds0.DoseUnits})"
-                )
-                return None
-
-        if not ds.DoseSummationType == "PLAN":
-            st.error(f"File {fh.name} is not a 'plan' dose")
-            return None
+            _validate_comparison_to_initial(ds, ds0)
+        except ValueError:
+            st.error(
+                f"When trying to load {fh.name} and compare it to {files[0].name} "
+                "the following error occurred:"
+            )
+            raise
 
         datasets.append(ds)
 
     return datasets
 
 
+def _load_dicom_file(fh: BinaryIO):
+    try:
+        ds = pydicom.dcmread(fh)
+    except pydicom.errors.InvalidDicomError as e:
+        raise ValueError(f"'{fh.name}' is not a valid DICOM file") from e
+
+    try:
+        _validate_dicom_dataset(ds)
+    except ValueError:
+        st.error(f"When trying to load {fh.name} the following error occurred:")
+
+        raise
+
+    return ds
+
+
+def _validate_dicom_dataset(ds):
+    if ds.Modality != "RTDOSE":
+        raise ValueError("File is not a valid DICOM RT Dose file")
+
+    if ds.DoseSummationType != "PLAN":
+        raise ValueError("File is not a 'plan' dose")
+
+
+def _validate_comparison_to_initial(ds, ds0):
```

Comment:
I've just worked out we need to require that `ds.DoseUnits = 'GY'`. We can't really add relative doses easily - they'll have different norm points. Or at least the current implementation would fail to do so correctly.

With that in mind, this function isn't necessary and a simple `ds.PatientId == ds0.PatientId` would suffice.